pkg/syncer/shared/namespace.go

```diff
@@ -20,40 +20,75 @@ import (
 	"crypto/sha256"
 	"encoding/json"
 	"fmt"
+	"strings"
 
 	"github.com/kcp-dev/logicalcluster"
+	"github.com/martinlindhe/base36"
 )
 
 const (
-	NamespaceLocatorAnnotation = "kcp.dev/namespace-locator"
+	NamespaceAnnotation       = "workload.kcp.dev/namespace"
+	WorkspaceAnnotation       = "workload.kcp.dev/workspace"
+	SyncerWorkspaceAnnotation = "workload.kcp.dev/syncer"
 )
 
 // NamespaceLocator stores a logical cluster and namespace and is used
 // as the source for the mapped namespace name in a physical cluster.
 type NamespaceLocator struct {
-	LogicalCluster logicalcluster.Name `json:"logical-cluster"`
-	Namespace      string              `json:"namespace"`
+	Workspace       logicalcluster.Name `json:"logical-cluster"`
+	SyncerWorkspace logicalcluster.Name `json:"syncer-workspace"`
+	Namespace       string              `json:"namespace"`
 }
 
 func LocatorFromAnnotations(annotations map[string]string) (*NamespaceLocator, error) {
-	annotation := annotations[NamespaceLocatorAnnotation]
```

Comment:
What's the benefit of moving this out of one annotation?

Reply:
I've kept the annotation for now, after playing with the idea, I don't see too much value.